### PR TITLE
extra exercise after basics

### PR DESCRIPTION
**Pull request by DaanVanHauwermeiren**
Additional exercise after `01-basics.jl`.
This is part of #66, related to #68, also see #44

The exercise is based on day3 of AoC 2020. The idea is to implement the solution of Michiel as the base solution. And my solution as an extremely convoluted one to give a sneak peak into the other material of the course.

### CODE REVIEW

**Comment by Beramos** [comment]
Very nice

**Comment by Beramos** [comment]
Looks good. Pls fix my remark and you can merge this.

File: notebooks/day1/01-basics-extra.jl [hunk 1]
```diff
@@ -0,0 +1,402 @@
...
+slopestyles = (
+    (1, 1), 
+    (3, 1), 
+    (5, 1), 
+    (7, 1), 
+    (1, 2), 
+)
+
+# ╔═╡ e0569568-63ed-11eb-33b9-51a8b179292c
+prod([goingdowntheslope(direction..., slope) for direction in slopestyles])
```
Comment (Beramos) [inline notebooks/day1/01-basics-extra.jl:363]: This throws a dimension error....

Comment (DaanVanHauwermeiren) [reply notebooks/day1/01-basics-extra.jl:363]: This does not throw an error for me.

Comment (Beramos) [reply notebooks/day1/01-basics-extra.jl:363]: I also don't get an error anymore. :+1:

**Pull request merged**